lnd.go

```diff
@@ -249,6 +249,18 @@ func Main(cfg *Config, lisCfg ListenerCfg, implCfg *ImplementationCfg,
 		defer runtimePprof.StopCPUProfile()
 	}
 
+	// Write memory profile if requested.
+	if cfg.MemProfile != "" {
+		f, err := os.Create(cfg.MemProfile)
+		if err != nil {
+			return mkErr("unable to create mem profile: %v", err)
+		}
+		defer func() {
```

Comment:
Looking at the `godoc`, looks like we should call `runtime.GC()` before writing the mem profile? https://pkg.go.dev/runtime/pprof

```go
func main() {
    flag.Parse()
    if *cpuprofile != "" {
        f, err := os.Create(*cpuprofile)
        if err != nil {
            log.Fatal("could not create CPU profile: ", err)
        }
        defer f.Close() // error handling omitted for example
        if err := pprof.StartCPUProfile(f); err != nil {
            log.Fatal("could not start CPU profile: ", err)
        }
        defer pprof.StopCPUProfile()
    }

    // ... rest of the program ...

    if *memprofile != "" {
        f, err := os.Create(*memprofile)
        if err != nil {
            log.Fatal("could not create memory profile: ", err)
        }
        defer f.Close() // error handling omitted for example
        runtime.GC() // get up-to-date statistics
        if err := pprof.WriteHeapProfile(f); err != nil {
            log.Fatal("could not write memory profile: ", err)
        }
    }
}
```
```

Reply:
https://pkg.go.dev/runtime/pprof#Profile

Reply:
Pushed call to `runtime.GC()`. Also, added error handling for the missing cases of the mem and cpu profiles.

https://github.com/lightningnetwork/lnd/compare/d709f89d0d5c85b6f870dc6ccb39d06bd4bbef29..e6e89ac450c25744512273f0bbe5f9f7b9dbcc95